syncer/syncer.go

```diff
@@ -165,8 +176,18 @@ func (s *syncer) Sync(callback func(*types.FullBlock) bool) error {
 	skipList := make(map[peer.ID]bool)
 
 	for {
+		s.lock.RLock()
+		blockTimeout := s.blockTimeout
+		s.lock.RUnlock()
+
 		// Wait for a new event to arrive
-		<-s.newStatusCh
+		select {
+		case <-s.newStatusCh:
+			s.logger.Debug("new peer status arrived, start syncing")
+		case <-time.After(blockTimeout):
```

Comment:
```suggestion
		case <-time.After(blockTimeout * 2):
```
I think we should wait a little bit longer because a new block will arrive only for at least 2 seconds. In addition we should consider what to do if a new round started :(
So we have to find a way to be smart and to know how long to wait, otherwise we are going to ping other peers without a reason.

Reply:
`blockTimeout` is `block time * 3`, and I've made it even longer here by multiplying it with 2, so it is `block time * 6`. We can always increase this.

Since this action is not something we expect to happen often, plus this timeout is pretty large, I would say, it won't have much influence if we ping the peers every `block time * 6`. But, like I said, we can increase this.

Reply:
Ok, I updated a comment, no need to add additional *2 in this place. It's only a question what to do in the case of higher rounds. I guess we can do nothing since we don't know how long will take for current block to be minded, i.e. what round is in progress

Reply:
Yes, exactly. If it gets into multiple rounds, there is nothing we can do.